invenio/modules/workflows/static/js/workflows/selection.js

```diff
@@ -140,14 +134,43 @@ define(
         }
       };
 
-      this.removeElementFromIDs = function(id) {
+      this.removeSentElements = function(ev, removedIDs) {
+        this.removeElementFromIDs(removedIDs);
+      };
+
+      this.removeElementFromIDs = function(removedIDs) {
         var idArray = this.attr.selectedIDs;
-        if (idArray.length > 0) {
-          var index = idArray.indexOf(id);
-          idArray.splice(index, 1);
+
+        removedIDs.ids.forEach(function(id) {
+          var index = idArray.indexOf(id.toString());
+          if (index > -1) {
+           idArray.splice(index, 1);
+          }
+        });
+
+        console.log("Remaining: " + this.attr.selectedIDs);
+      };
+
+      this.setCheckboxes = function(ev, data) {
+        var idArray = this.attr.selectedIDs;
+
+        idArray.forEach(function(id) {
+          if ($("input[value=" + id +"]").prop('checked') == false) {
+            $("input[value=" + id +"]").prop('checked', true);
+          }
+        });
+
+        if (idArray.length == 0 || (data.pagination.per_page != idArray.length && data.pagination.pages > 1)) {
+          $(this.attr.selectAllSelector).prop('checked', false);
+        } else if (data.pagination.pages == 1 && data.pagination.total_count == idArray.length) {
+          $(this.attr.selectAllSelector).prop('checked', true);
         }
       };
 
+      this.debugMe = function() {
```

Comment:
To declutter things, lets remove the debug code. This and the line below (190 with the `this.on(document, "showids" ...`